crates/builder/build.rs

```diff
@@ -0,0 +1,45 @@
+use std::env;
+use std::fs;
+use std::path::Path;
+use std::process::{Command, ExitStatus};
+use std::str;
+
+// Checks if the maybe_uninit_write_slice feature can be enabled
+const MAYBE_UNINIT_WRITE_SLICE_PROBE: &str = r#"
+#![feature(maybe_uninit_write_slice)]
+#![allow(dead_code)]
+
+use std::mem::MaybeUninit;
+
+pub fn test() {
+    MaybeUninit::write_slice(
+        [MaybeUninit::<usize>::uninit()].as_mut_slice(),
+        [1_usize].as_slice()
+    );
+}
+"#;
+
+fn main() {
+    match compile_probe(MAYBE_UNINIT_WRITE_SLICE_PROBE) {
+        Some(status) if status.success() => {}
+        _ => println!("cargo:rustc-cfg=no_maybe_uninit_write_slice"),
```

Comment:
Maybe for readability, we can put do

```
fn enable_compat_for_feature(feature_name: &str) {
    println!("cargo:rust-cfg=no_{feature_name}");
}
```

and in the `Err` case:

` _ => enable_compat_for_feature("maybe_uninit_write_slice")`